pkgs/by-name/im/imsprog/package.nix

```diff
@@ -0,0 +1,54 @@
+{ lib, stdenv, fetchFromGitHub, cmake, makeWrapper, pkg-config, bash, gnome, libusb1, qt5, wget }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "imsprog";
+  version = "1.4.1";
+
+  src = fetchFromGitHub {
+    owner = "bigbigmdm";
+    repo = "IMSProg";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-eF6TGlFEnMgMw1i/sfuXIRzNySVZe7UTKVHSIqJ+cUs=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+    pkg-config
+    qt5.wrapQtAppsHook
+    qt5.qttools
+  ];
+
+  buildInputs = [
+    bash # for patching the shebang in bin/IMSProg_database_update
+    libusb1
+    qt5.qtbase
+    qt5.qtwayland
+  ];
+
+  # change default hardcoded path for chip database file, udev rules et al
+  postPatch = ''
+    while IFS= read -r -d "" file ; do
+      substituteInPlace "$file" \
+        --replace-quiet '/usr/bin/' "$out/bin/" \
+        --replace-quiet '/usr/lib/' "$out/lib/" \
+        --replace-quiet '/usr/share/' "$out/share/"
+    done < <(grep --files-with-matches --null --recursive '/usr/' .)
+  '';
+
+  preFixup = ''
+    wrapProgram $out/bin/IMSProg_database_update \
```

Comment:
Don't use this as it will wrap the program twice. Use `qtWrapperArgs`: https://nixos.org/manual/nixpkgs/unstable/#qt-runtime-dependencies

Reply:
`IMSProg_database_update` is a bash script, not a QT application

Reply:
To add to that: the last info box on https://nixos.org/manual/nixpkgs/stable/#qt-runtime-dependencies says:

> wrapQtAppsHook ignores files that are non-ELF executables.

However, an `ls` reveals that there is stil double-wrapping:

```console
$ ls -la result/bin/
.r-xr-xr-x 735k root  1 Jan  1970 .IMSProg-wrapped
.r-xr-xr-x 4,1k root  1 Jan  1970 .IMSProg_database_update-wrapped
.r-xr-xr-x  16k root  1 Jan  1970 .IMSProg_database_update-wrapped_
.r-xr-xr-x 160k root  1 Jan  1970 .IMSProg_editor-wrapped
.r-xr-xr-x  16k root  1 Jan  1970 IMSProg
.r-xr-xr-x  16k root  1 Jan  1970 IMSProg_database_update
.r-xr-xr-x  16k root  1 Jan  1970 IMSProg_editor
```

The culprit likely is that our new `wrap*AppsHook` implementation itself creates an ELF, that then gets wrapped by wrapQtAppsHook. Changing my custom wrapper from `preFixup` to `postFixup` solves the issue of double wrapping.

Reply:
Fixed in 5ce8db90a86534f5dadfdeeee466dfe4c4e6619c